src/lib/snapper/snapshot_dbus.rb

```diff
@@ -0,0 +1,214 @@
+require 'yast'
+require 'dbus'
+
+module Yast2
+  module Snapper
+    class SnapshotDBus
+      
+      include Yast::Logger
+
+      TYPE_INT_TO_SYMBOL = {
+        0 => :SINGLE,
+        1 => :PRE,
+        2 => :POST
+      }
+      def initialize(context)
+        @context = context
+      end
+
+      class << self
+        
+        def all(configs)
+          configs = list_configs if !configs
+          snapshots = [configs].flatten.each_with_object([]) do |config, a|
+            list_snapshots(config).each do |snapshot|
+              next if snapshot[0] == 0
+              s = {
+                "num" => snapshot[0],
+                "type" => TYPE_INT_TO_SYMBOL[snapshot[1]],
+                "pre_num" => snapshot[2],
+                "date" => Time.at(snapshot[3]),
+                "uid" => snapshot[4],
+                "description" => unescape(snapshot[5]),
+                "cleanup" => unescape(snapshot[6]),
+                "userdata" => unescape(snapshot[7])
+              }
+              a << Snapshot.new(s["num"], s["type"], s["pre_num"], s["date"],
+                s["uid"], s["description"], s["cleanup"], s["userdata"], config, self) 
```

Comment:
I extract it from list_snapshots because was it original place in SnapperDbus, and leave there just  because i want to discuss about the constructor.  

I prefer initialize(attrs = nil) or initialize(num, type, uid, *attrs) doing all the logic of validation inside or at least taking care of defaults values.